tasks/docker-swarm.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: Init a new swarm
+  community.docker.docker_swarm:
+    listen_addr: "{{ docker_swarm_addr }}:{{ docker_swarm_port }}"
+    advertise_addr: "{{ docker_swarm_addr }}"
+    subnet_size: "{{ docker_swarm_default_addr_pool_mask_length }}"
+    default_addr_pool: "{{ docker_swarm_default_addr_pool }}"
+  when: inventory_hostname == groups['docker_swarm_managers'][0]
+
+- name: Get the join-tokens.
+  community.docker.docker_swarm:
+  register: docker_info
+  delegate_to: "{{ groups['docker_swarm_managers'][0] }}"
+  when: inventory_hostname != groups['docker_swarm_managers'][0]
+
+- name: Declare the address of the first Swarm manager as a fact.
+  set_fact:
+    docker_manager_address: "{{ docker_swarm_addr }}:{{ docker_swarm_port }}"
+  when: "inventory_hostname == groups['docker_swarm_managers'][0]"
+
+- name: Distribute the fact containing address of the first Swarm manager.
+  set_fact:
+    docker_manager_address: "{{ hostvars[groups['docker_swarm_managers'][0]]['docker_manager_address'] }}"
+  when: "inventory_hostname != groups['docker_swarm_managers'][0]"
```

Comment:
You do not need 2 tasks here. Use this instead:

`docker_manager_address: "{{ hostvars[groups['docker_swarm_managers'][0]]['docker_swarm_addr'] }}:{{ hostvars[groups['docker_swarm_managers'][0]]['docker_swarm_port'] }}"`

Reply:
The task includes an option with an undefined variable. The error was: 'ansible.vars.hostvars.HostVarsVars object' has no attribute 'docker_swarm_addr'

Reply:
Then there is no {{ docker_swarm_port }} defined in previous one

Reply:
Variables that specified in role and not specified in hostvars/groupvars are not included in hostvars variable.